Task 10 data upload

Aureliasusu opened the pull request:
1.implement profile edition feature, account deletion feature and render profile photo in front page
<img width="1461" height="899" alt="profile update dans account deletion" src="https://github.com/user-attachments/assets/250c3e6c-2002-4427-950e-67fb593fbe3c" />

2.implement multi-form data upload feature and render files list
<img width="1834" height="628" alt="successfully implemented upload feature" src="https://github.com/user-attachments/assets/3b995c99-681c-4fc7-902a-cd3e5a88547b" />

3.refine register page
<img width="1337" height="797" alt="register page - membership update" src="https://github.com/user-attachments/assets/15173251-f7cf-49c2-abca-71efb63cbbdc" />

4. refine my organization panel
<img width="816" height="751" alt="refine my organizations penal to show details" src="https://github.com/user-attachments/assets/2163be7f-5d86-42e1-a949-99944e6a38ff" />

Aureliasusu committed:
new welcomepage and profile edition & photo upload

Aureliasusu committed:
implement upload data feature/ render  files list

victiny23 reviewed (changes requested):
User information should not be in the code base

Aureliasusu committed:
🔒 SECURITY: Implement comprehensive data protection

- Add comprehensive .gitignore rules to exclude all user data
- Remove database files and user uploads from repository
- Create DataInitializationConfig for sample data only
- Add secure-data-setup.sh script for safe development setup
- Create SECURITY_GUIDE.md with best practices
- Ensure no real user data is committed to open repository

This ensures the open GitHub repository contains only code and sample data,
while all user data (databases, uploads, profile photos) is properly excluded.

Aureliasusu committed:
🚀 PRODUCTION: Add secure production deployment configuration

- Add application-prod.yml with external database and cloud storage config
- Create comprehensive PRODUCTION_DEPLOYMENT.md guide
- Configure PostgreSQL, Redis, and cloud storage for production
- Include Docker Compose production setup
- Add security checklist and monitoring guidelines
- Provide backup and emergency procedures

This ensures production deployments use external databases and cloud storage,
keeping the open repository completely free of user data.

Aureliasusu committed:
🚨 SECURITY: Add immediate action guide for data protection

- Create IMMEDIATE_SECURITY_ACTIONS.md with urgent steps
- Provide git filter-branch commands to remove user data from history
- Include verification checklist for repository health
- Add emergency procedures for data breach response
- Ensure complete security compliance for open repository

This document provides immediate steps to secure the repository
and prevent future exposure of user data.

Aureliasusu committed:
lab membership system & register page file page update

Aureliasusu committed:
refine register page and user management page
